### PR TITLE
DIP25: make phobos work with it

### PR DESCRIPTION
**Pull request by WalterBright**
Needs https://github.com/D-Programming-Language/dmd/pull/4357

### CODE REVIEW

File: std/algorithm/package.d [hunk 1]
```diff
@@ -17,25 +17,25 @@ $(TR $(TH Category) $(TH Submodule) $(TH Functions)
 $(TR $(TDNW Searching)
      $(TDNW $(SUBMODULE searching))
      $(TD
-        $(SUBREF searching, all)
```
Comment (Orvid) [inline std/algorithm/package.d]: Assuming my sanity is correct, shouldn't these changes normally be in their own commit? (they are just whitespace changes due to there being extra blanks at the end of the line)

Comment (WalterBright) [inline std/algorithm/package.d]: It isn't worth the bother. (the original committer had failed to run `tolf` before committing)

**Comment by schuetzm** [comment]
This needs to go into its own branch, not master.

**Comment by andralex** [comment]
Nice. @schuetzm how do you mean that?

**Comment by schuetzm** [comment]
As far as I understand the situation, this (together with other modifications) will be an experimental implementation of DIP25. We don't yet know when it will be ready, and not even whether it will work out right at all.

Therefore, wouldn't it be better to keep it away from master until we know it's in a state where it can be released? Otherwise, we'll have a half-finished feature in our next release, or at least we would need to revert it in the release branch. (The same for DMD, of course.)

**Comment by WalterBright** [comment]
This is why it is only enabled with the `-dip25` switch. Putting it in a separate branch means that nobody will ever try it out.

**Comment by andralex** [comment]
@schuetzm I think we're good there. Thanks!
